### ISSUE TITLE
Multiple H1 tags & Programmatic Updates for the Registry

### ISSUE THREAD

**Issue by Brant-Soler** [pulumi/registry#646]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

I would like to propose a few changes to the H1 system within the /registry/. I'm going to be using the Kubernetes page as a reference but this actually applies to all of the templates section by section. The easiest sections are "How-to Guides" and "Installation & Configuration", I will talk about these first, the last section "API Docs" is going to be more of a challenge, and I will be covering it separately.

If you know of or find sections where these template changes will not work or create non-human English headings, speak up! Create a comment so we can work through the best solution!

_Side note: While I am not a developer, I have a fair understanding of developer language. Please forgive misused words and suggest a correction for clarity if there are any verbal or logical errors below._

For the How-to Guides sections:

[This page needs an addition to the H1](https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/) of "How-to Guides" at the end of "Kubernetes" as this page is a listing of the guides. **This directory page is also missing the ability to edit or request changes to this page which should live on the right side.** (Susan, this is why you've been added.)

The page titles for this section are already perfect matches to the addressing & soon to the only H1 on the page. Google looks for parity between address, title & H1 as a major page level quality score. Here is how this example looks to Google:

Address: /kubernetes/how-to-guides/
Title: Kubernetes How-to Guides | Pulumi
H1: Kubernetes How-to Guides <-- This is the updated state.

[This page includes two H1 tags](https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/gke/) when there should only be one, "Kubernetes" at the top and "Google Kubernetes Engine (GKE) Tutorial" at the top of the body section. 

For How-to Guides pages, the second H1 "Google Kubernetes Engine (GKE) Tutorial" should replace the top H1 "Kubernetes". This way the title of the How-to Guide is the H1 for the page.

The address needs an update and 301 redirect to match the H1 & Title. **We can wait to make this update as a more complete set of updates & 301 redirects**, though if we include it in this process, here is the update:

From: /kubernetes/how-to-guides/gke/
To: /kubernetes/how-to-guides/google-kubernetes-engine-gke-tutorial/

The page titles for this section are already perfect matches to the addressing & soon to the only H1 on the page. Google looks for parity between address, title & H1 as a major page level quality score. Here is how this example looks to Google:

Address: /kubernetes/how-to-guides/gke/
Optimal address: /kubernetes/how-to-guides/google-kubernetes-engine-gke-tutorial/
Title: Google Kubernetes Engine (GKE) Tutorial | Pulumi
H1: Google Kubernetes Engine (GKE) Tutorial <-- This is the updated state.

**There is no change to page titles here.**

[This page needs an addition to the H1](https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/) of "How-to Guides" at the end of "Kubernetes" as this page is a listing of the guides. This directory page is also missing the ability to edit or request changes to this page which should live on the right side.

The page titles for this section are already perfect matches to the addressing & soon to the only H1 on the page. Google looks for parity between address, title & H1 as a major page level quality score. Here is how this example looks to Google:

Address: /kubernetes/how-to-guides/
Title: Kubernetes How-to Guides | Pulumi
H1: Kubernetes How-to Guides <-- This is the updated state.

**There is no change to page titles here.**

For the Installation & Configuration sections:

The H1 on this page should have "Installation & Configuration" appended to the end to read "Kubernetes Installation & Configuration".

The page title needs to be updated to "Kubernetes Installation & Configuration". Here is how this example will look to Google:

Address: /kubernetes/installation-configuration/
Title: Kubernetes Installation & Configuration | Pulumi
H1: Kubernetes Installation & Configuration

### Affected area/feature

[Google Kubernetes Engine (GKE) Tutorial](https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/gke/)
[Kubernetes How-to Guides](https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/)
[Kubernetes Installation & Configuration](https://www.pulumi.com/registry/packages/kubernetes/installation-configuration/)

**Comment by susanev** [comment]
hi @Brant-Soler thx for writing this up ill take a look, but id appreciate if you did not assign me to issues, ill assign myself if/when i start actively working on it.

**Comment by susanev** [comment]
some ux notes

- even though each tab was a built out as a separate page that is not how it is presented to the user, the h1 is above the tabs and describes the content for the entire package, i do not think we should change the h1 as people select the tabs on this page; if someone would like to propose a redesign to this page, id be happy to give feedback
- we should absolutely always have exactly one h1 on a page, and we should def fix anywhere where that is not true

**Comment by Brant-Soler** [comment]
> * even though each tab was a built out as a separate page that is not how it is presented to the user, the h1 is above the tabs and describes the content for the entire package, i do not think we should change the h1 as people select the tabs on this page; if someone would like to propose a redesign to this page, id be happy to give feedback; alternatively, it might make sense to explore using anchors instead fo distinct URLs when navigating with the tabs

Thank you Susan, I'm confused about your first point. The separate pages are linked at the Affected Area/Feature portion of my post. If the page transitioned between sections without separate URLs, I would be recommending a different approach. Each of these pages is registering as independent in the Pulumi Registry /registry/packages/kubernetes/how-to-guides/gke/, the breadcrumbs also represent the nested directory structure. 

Your third point is spot on with another item on my "how to approach" list: modules where there are no How-to guides such as these:
https://www.pulumi.com/registry/packages/digitalocean/how-to-guides/ 
https://www.pulumi.com/registry/packages/confluent/how-to-guides/ 
https://www.pulumi.com/registry/packages/kong/how-to-guides/

**Comment by susanev** [comment]
i think maybe we should zoom sometime soon and talk through it to get aligned on what you want to happen with the headings

**Comment by susanev** [comment]
to follow-up here. the user experience team is going to work on a redesign of the header on these pages to move away from the package parent idea. this will allow us to solve the SEO issues without introducing accessibility issues. we are not pulling this in as emergent because Christian needs time to investigate how to fix the core eng issue and has a number of other projects he is currently working on.

**Comment by susanev** [comment]
@Brant-Soler can you provide more details on what emergent SEO issues Christian is solving? i think we may have left the meeting with different understandings. this is what i left the meeting thinking..

- Christian is going to focus on solving the multiple h1 issues on a single page
- Next iteration the ux design team is going to work on a header redesign in order to accommodate the SEO needs of including overview, installation & configuration, API docs, and how-to guides in the h1 without introducing accessibility issues

**Comment by Brant-Soler** [comment]
Absolutely, the emerging SEO issue, detailed at the top of this thread, is:

- The multitude of identical primary H1 tags within each package within the /registry/. The secondary issue is the presence of second H1 tags on some pages which can be addressed at a later time if the lift is too large.

As we discussed, the H1 for all of my example pages below is "Kubernetes". This type of technical SEO error suppresses performance, these pages are high performers which are being held back from their potential. If you look at the addresses for each of these pages, you will note a strong correlation with what should be the H1 tags:

https://www.pulumi.com/registry/packages/kubernetes/ - This H1 should be "Kubernetes Package"
https://www.pulumi.com/registry/packages/kubernetes/installation-configuration/ - This H1 should be "Kubernetes Package Installation & Configuration"
https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/ - This H1 should be "Kubernetes Package How-to Guides"
https://www.pulumi.com/registry/packages/kubernetes/how-to-guides/guestbook/ - This H1 should be "Kubernetes Guestbook App with Redis and Nginx"

There are more examples of this emergent SEO issue, along with recommendations to correct them, in the GitHub issue [647](https://github.com/pulumi/registry/issues/647).

Christian is currently investigating the lift involved with correcting this technical error against the lift involved in the UX design team working on a header redesign in order to accommodate this SEO need.

**Comment by susanev** [comment]
thx @cnunciato yes i agree. i do not want us to change the headings above the navigation until we redesign this page, as it will lead to both usability and accessibility issues which i do not think we should sacrifice to fix the SEO issue.
